### PR TITLE
Resolve ZHVJQQM9 933160 bypass

### PR DESCRIPTION
**Pull request by 53cur3M3**

### CODE REVIEW

File: util/regexp-assemble/data/933160.data [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
 
 ##!+ i
-##!^ \b
-##!$ (?:\s|/\*.*\*/|//.*|#.*|\")*\(.*\)
+##! optional opening bracket before command
+##!^ \b\(?
+##! optional qoutes
+##!^ ['\"]*
+
+##! 933160-40 test fails without following suffix:
```
Comment (theseion) [inline util/regexp-assemble/data/933160.data:13]:
```suggestion
```

File: util/regexp-assemble/data/933160.data [hunk 2]
```diff
@@ -1,9 +1,25 @@
 ##! Please refer to the documentation at 
 ##! https://coreruleset.org/docs/development/regexp_assemble/.
+##! \b
```
Comment (theseion) [inline util/regexp-assemble/data/933160.data:3]: Remove these commented bits.

File: util/regexp-assemble/data/933160.data [hunk 3]
```diff
@@ -1,9 +1,25 @@
...
 ##!+ i
-##!^ \b
-##!$ (?:\s|/\*.*\*/|//.*|#.*|\")*\(.*\)
+##! optional opening bracket before command
+##!^ \b\(?
+##! optional qoutes
+##!^ ['\"]*
+
+##! 933160-40 test fails without following suffix:
+##!$ (?:\s|/\*.*\*/|//.*|#.*|\")*
```
Comment (theseion) [inline util/regexp-assemble/data/933160.data:14]:
```suggestion
##! match comments: `/*...*/`, `//...`, `#...`
##!$ (?:/\*.*\*/|//.*|#.*
##! match white space and quotes
##!$ \s|\")*
```

File: util/regexp-assemble/data/933160.data [hunk 4]
```diff
@@ -1,9 +1,25 @@
 ##! Please refer to the documentation at 
 ##! https://coreruleset.org/docs/development/regexp_assemble/.
+##! \b
+##! (?:\s|/\*.*\*/|//.*|#.*|\")*\(.*\)
+##!                       \)?\s*\(.*\)
 
 ##!+ i
-##!^ \b
-##!$ (?:\s|/\*.*\*/|//.*|#.*|\")*\(.*\)
+##! optional opening bracket before command
```
Comment (theseion) [inline util/regexp-assemble/data/933160.data:8]:
```suggestion
##! optional opening parenthesis before command
```

File: util/regexp-assemble/data/933160.data [hunk 5]
```diff
@@ -1,9 +1,25 @@
...
+##!^ \b\(?
+##! optional qoutes
```
Comment (theseion) [inline util/regexp-assemble/data/933160.data:10]:
```suggestion
##! optional quotes
```

File: util/regexp-assemble/data/933160.data [hunk 6]
```diff
@@ -1,9 +1,25 @@
...
+##!^ ['\"]*
+
+##! 933160-40 test fails without following suffix:
+##!$ (?:\s|/\*.*\*/|//.*|#.*|\")*
+
+##! optional qoutes
```
Comment (theseion) [inline util/regexp-assemble/data/933160.data:16]:
```suggestion
##! optional quotes
```

File: util/regexp-assemble/data/933160.data [hunk 7]
```diff
@@ -1,9 +1,25 @@
...
+##!$ ['\"]*
+##! optional closing bracket after command
```
Comment (theseion) [inline util/regexp-assemble/data/933160.data:18]:
```suggestion
##! optional closing parenthesis after command
```

**Review by theseion — CHANGES_REQUESTED** [review]

File: util/regexp-assemble/data/933160.data (hunk 2 above)
Comment (53cur3M3) [reply util/regexp-assemble/data/933160.data]: Previous commented prefix and suffix removed.

File: tests/regression/tests/REQUEST-933-APPLICATION-ATTACK-PHP/933160.yaml [hunk 8]
```diff
@@ -645,3 +645,51 @@ tests:
...
+              Host: localhost
+              User-Agent: OWASP ModSecurity Core Rule Set
+              Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
+            method: POST
+            port: 80
+            uri: /
+          output:
+            log_contains: id "933160"
+  - test_title: 933160-42
+    desc: system in braces - whoami in braces
```
Comment (theseion) [inline tests/regression/tests/REQUEST-933-APPLICATION-ATTACK-PHP/933160.yaml:665]:
```suggestion
    desc: system in braces - whoami in parentheses
```

File: tests/regression/tests/REQUEST-933-APPLICATION-ATTACK-PHP/933160.yaml [hunk 9]
```diff
@@ -645,3 +645,51 @@ tests:
             uri: /?foo=return%22system%22%28xyz%29.s
```
Comment (theseion) [inline tests/regression/tests/REQUEST-933-APPLICATION-ATTACK-PHP/933160.yaml:645]: Remove the comment on the line above.

**Review by theseion — CHANGES_REQUESTED** [review]

File: tests/regression/tests/REQUEST-933-APPLICATION-ATTACK-PHP/933160.yaml [hunk 10]
```diff
@@ -645,3 +645,51 @@ tests:
...
+            dest_addr: 127.0.0.1
+            headers:
+              Host: localhost
+              User-Agent: OWASP ModSecurity Core Rule Set
+              Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
+            method: POST
+            port: 80
+            uri: /
+          output:
+            log_contains: id "933160"
```
[the remark this answers is not in the document]
Comment (53cur3M3) [reply tests/regression/tests/REQUEST-933-APPLICATION-ATTACK-PHP/933160.yaml:695]: Done

File: tests/regression/tests/REQUEST-933-APPLICATION-ATTACK-PHP/933160.yaml (hunk 9 above)
Comment (53cur3M3) [reply tests/regression/tests/REQUEST-933-APPLICATION-ATTACK-PHP/933160.yaml:644]: Done

**Comment by 53cur3M3** [comment]
@karelorigin please can you review rebased rule, as there was a merge conflict with your change.

**Review by theseion — APPROVED** [review]

**Comment by karelorigin** [comment]
Doesn't look like there are any issues after solving the merge conflict :+1:

**Pull request merged**